Sources/LeafKit/LeafSource/NIOLeafFiles.swift

```diff
@@ -1,13 +1,89 @@
-/// Default implementation of `LeafFiles` adhering object that provides a non-blocking file reader
-/// to `LeafRenderer`
-public struct NIOLeafFiles: LeafFiles {
-    let fileio: NonBlockingFileIO
+import Foundation
 
-    public init(fileio: NonBlockingFileIO) {
+/// Reference and default implementation of `LeafSource` adhering object that provides a non-blocking
+/// file reader for `LeafRenderer`
+///
+/// Default initializer will
+public struct NIOLeafFiles: LeafSource {
+    public struct Limit: OptionSet {
+        public let rawValue: Int
+        public init(rawValue: Int) {
+            self.rawValue = rawValue
+        }
+        
+        /// Require any referenced file have an extension
+        public static let requireExtensions = Limit(rawValue: 1 << 0)
+        /// Require any referenced file end in `.leaf`
+        public static let onlyLeafExtensions = Limit(rawValue: 1 << 1)
+        /// Limit access to inside configured sandbox directory
+        public static let toSandbox = Limit(rawValue: 1 << 2)
+        /// Limit access to visible files/directories
+        public static let toVisibleFiles = Limit(rawValue: 1 << 3)
+        
+        public static let `default`: Limit = [.toSandbox, .toVisibleFiles, .requireExtensions]
+        public static let dirLimited: Limit = [.toSandbox, .toVisibleFiles]
+    }
+    
+    let fileio: NonBlockingFileIO
```

Comment:
Should the casing for this property's name be changed to `fileIO`?

Reply:
Maybe? 🙃 I didn't name it!